src/components/smartSearch/filters/TimeFrame.tsx

```diff
@@ -0,0 +1,152 @@
+import DateFnsUtils from '@date-io/date-fns';
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { MenuItem, TextField, Typography } from '@material-ui/core';
+import { FormattedMessage as Msg, useIntl } from 'react-intl';
+import { useEffect, useState } from 'react';
+
+enum RANGES {
+    EVER='ever',
+    FUTURE='future',
+    BEFORE_TODAY='beforeToday',
+    BEFORE_DATE='beforeDate',
+    AFTER_DATE='afterDate',
+    BETWEEN='between',
+    LAST_FEW_DAYS='lastFew',
+}
+
+interface TimeFrameProps {
+    onChange: (range: {after?: string; before?: string}, canSubmit: boolean) => void;
+}
+
+const TimeFrame = ({ onChange }: TimeFrameProps): JSX.Element => {
+    const intl = useIntl();
+    const [selected, setSelected] = useState<RANGES>(RANGES.EVER);
+    const [before, setBefore] = useState<MaterialUiPickersDate  | null>(null);
+    const [after, setAfter] = useState<MaterialUiPickersDate  | null>(null);
+    const [numDays, setNumDays] = useState(30);
+
+    useEffect(() => {
+        onChange({}, false);
+        if (selected ===  RANGES.EVER) {
+            onChange({}, true);
+        }
+        if (selected === RANGES.BEFORE_TODAY) {
+            onChange({ before: 'now' }, true);
+        }
+        if (selected === RANGES.FUTURE) {
+            onChange({ after: 'now' }, true);
+        }
+        if (selected === RANGES.LAST_FEW_DAYS) {
+            onChange({ after: `-${numDays}d` }, true);
+        }
+        if (selected === RANGES.BEFORE_DATE && before) {
+            onChange({ before: before.toISOString().slice(0, 10) }, true);
+        }
+        if (selected === RANGES.AFTER_DATE && after) {
+            onChange({ after: after.toISOString().slice(0, 10) }, true);
+        }
+        if (selected === RANGES.BETWEEN && after && before) {
+            onChange({
+                after: after.toISOString().slice(0, 10),
+                before: before.toISOString().slice(0, 10),
+            }, true);
+        }
+    }, [before, after, selected, numDays]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
disabled because I don't want to pass in onChange to the dependency array, otherwise I have to memoize the function in the parent component and I didn't want to do that, too messy.